addons/web/static/src/legacy/scss/bootstrap_review.scss

```diff
@@ -208,19 +208,42 @@ $o-navbar-nav-pills-link-border-radius: $nav-pills-border-radius !default;
     // body is white, the text will not be visible on modals with a white
     // background.
 
-    // Check if the body text color is light.
-    $-is-body-text-light: color-yiq($body-color) == $yiq-text-dark;
-
     @if $modal-content-color == null {
-        color: if(
-            $-is-body-text-light,
-            color-yiq($modal-content-bg, $light: $body-color, $dark: $yiq-text-dark),
-            color-yiq($modal-content-bg, $light: $yiq-text-light, $dark: $body-color)
-        );
+        color: adjust-body-text-color-for-background($modal-content-bg);
     }
-    @if $-is-body-text-light {
+    // Checks if the body text color is light.
+    @if color-yiq($body-color) == $yiq-text-dark {
         .text-muted {
             color: color-yiq($modal-content-bg, $light: $text-muted, $dark: mute-color($yiq-text-dark)) !important;
```

Comment:
This fails somehow. The modal's background normally matches the body, true,  but it's not always the case 😉 
In the following example I've kept the body to `#FFFFFF` (default), but I've customized the modal background color using variables.

```scss
$modal-content-bg: #4e0101;
```

![301218233-f8890ae1-3ae3-4635-a7b2-de800418027c](https://github.com/odoo/odoo/assets/8558351/3aaaf0a4-22fb-4d6e-89df-d19211749611)

Reply:
Hi @stefanorigano thanks for your review :+1: 
The point you talk about was already discussed here in the PR where this line was introduced https://github.com/odoo/odoo/pull/147347#discussion_r1445888643.
I'll let you reconsider your remark based on that. If you still think it's better to change that, I will do it :+1: .

Reply:
@bvr-odoo Why not checking that the `$text-muted` is actually the light one in the case of this line changing the `text-muted` rule? :) You would solve all problems no?

Reply:
@qsm-odoo  I think the right way to cover all cases is to do something like this, in the same way we did for modal content text color..

```css
    .text-muted {
        color: if(
            color-yiq($text-muted) == $yiq-text-dark,
            color-yiq($modal-content-bg, $text-muted, mute-color($yiq-text-dark)),
            color-yiq($modal-content-bg, mute-color($yiq-text-light), $text-muted)
        );
    }
 ```

Reply:
@qsm-odoo @stefanorigano I updated the PR to fix the case of `text-muted` in a dark modal.
I introduced a new `adjust-muted-text-color-for-background` function. Thanks for your remarks